ui/app/components/app/modals/new-account-modal/new-account-modal.component.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Button from '../../../ui/button/button.component'
+
+export default class NewAccountModal extends Component {
+
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    hideModal: PropTypes.func.isRequired,
+    newAccountNumber: PropTypes.number.isRequired,
+    onSave: PropTypes.func.isRequired,
+  }
+
+  state = {
+    alias: '',
+  }
+
+  onChange = e => {
+    this.setState({
+      alias: e.target.value,
+    })
+  }
+
+  onKeyPress = e => {
+    if (e.key === 'Enter' && this.state.alias) {
+      this.props.onSave()
+    }
+  }
+
+  render () {
+    const { t } = this.context
+
+    return (
+      <div className="new-account-modal">
+        <div className="new-account-modal__content">
+          <div className="new-account-modal__content__header">
+            {t('newAccount')}
+          </div>
+          <div className="new-account-modal__input-label">
+            {t('accountName')}
+          </div>
+          <input
+            type="text"
+            className="new-account-modal__input"
+            onChange={this.onChange}
+            onKeyPress={this.onKeyPress}
```

Comment:
Why is it weird? Allowing submissions of modals via `enter` keypress is also done in `metamask-extension/ui/app/components/app/modals/add-to-addressbook-modal/add-to-addressbook-modal.component.js` and I believe was an explicit specification in that case.

But it seems there is something about your emphasis on the word "_behind_" that I am missing?

Reply:
The modal remains open while the page loads behind it. Perhaps what is needed is for enter to close the modal as well

Reply:
Ah, it's because pressing enter calls `this.props.onSave()`, but pressing the 'Save' button calls `this.props.onSave(this.state.alias).then(this.props.hideModal)`. So I guess the solution is to create a helper method and call it in both cases.

Reply:
yes, good point

applied your suggestion in ad68be789